src/sinks/util/buffer/compression.rs

```diff
@@ -99,9 +100,34 @@ impl<'de> de::Deserialize<'de> for Compression {
                             if level.is_some() {
                                 return Err(de::Error::duplicate_field("level"));
                             }
-                            level = Some(match map.next_value::<usize>() {
-                                Ok(value) => value.to_string(),
-                                Err(_) => map.next_value::<&str>()?.to_string(),
+                            level = Some(match map.next_value::<Value>()? {
+                                Value::Number(level) => match level.as_u64() {
+                                    Some(value) if value <= 9 => value as usize,
+                                    Some(_) | None => {
+                                        return Err(de::Error::invalid_value(
+                                            de::Unexpected::Other("0, 1, 2, 3, 4, 5, 6, 7, 8 or 9"),
+                                            &self,
+                                        ))
+                                    }
+                                },
+                                Value::String(level) => match level.as_str() {
+                                    "none" => GZIP_NONE,
+                                    "fast" => GZIP_FAST,
+                                    "default" => GZIP_DEFAULT,
+                                    "best" => GZIP_BEST,
+                                    _ => {
+                                        return Err(de::Error::invalid_value(
+                                            de::Unexpected::Other("none, fast, default or best"),
+                                            &self,
+                                        ))
+                                    }
+                                },
+                                _ => {
+                                    return Err(de::Error::invalid_type(
+                                        de::Unexpected::Other("Number or String"),
```

Comment:
Same problem here. I'm not sure how to get an appropriate `Unexpected` variant here.